docs/advanced-topics/fuzz_introspector.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: default
+title: Fuzz Introspector
+parent: Advanced topics
+nav_order: 2
+permalink: /advanced-topics/fuzz-introspector/
+---
+
+# Fuzz Introspector
+{: .no_toc}
+
+For projects written in C/C++, Python and Java you can generate Fuzz
+Introspector reports to help guide the development of your fuzzing suit.
+These reports help to extract details about the fuzzing set up of your
+project with the goal of making it easier to improve the fuzzing set up.
+The Fuzz Introspector reports are generated automatically and uploaded
+to the cloud like code coverage reports, and you can also generate them
+locally using the OSS-Fuzz helper script.
+
+
+- TOC
+{:toc}
+---
+
+## Fuzz Introspector overview
+
+As soon as your project is run with ClusterFuzz (< 1 day), you can view the Fuzz
+Introspector report for your project.
+[Fuzz Introspector](https://github.com/ossf/fuzz-introspector) helps you
+understand your fuzzers' performance and identify any potential blockers.
+It provides individual and aggregated fuzzer reachability and coverage reports.
+You can monitor each fuzzer's static reachability potential and compare it
+against dynamic coverage and identify any potential bottlenecks.
+Fuzz Introspector can offer suggestions on increasing coverage by adding new
+fuzz targets or modify existing ones.
+Fuzz Introspector reports can be viewed from the [OSS-Fuzz
+homepage](https://oss-fuzz.com/) or through this
+[index](http://oss-fuzz-introspector.storage.googleapis.com/index.html).
+
+- [Fuzz Introspector documentation](https://fuzz-introspector.readthedocs.io/en/latest/)
+- [Fuzz Introspector source code](https://github.com/ossf/fuzz-introspector)
+- [OSS-Fuzz Fuzz Introspector reports](http://oss-fuzz-introspector.storage.googleapis.com/index.html)
+
+
+## Tutorials and guides
+
+The reports generated can be a lot to digest when first viewing them. The 
+[Fuzz Introspector documentation](https://fuzz-introspector.readthedocs.io/en/latest/)
+provides various user guides and tutorials rooted in OSS-Fuzz projects, which is
+a useful reference on how to make use of the reports.
+
+For ideas on how to use Fuzz Introspector, see [user guides](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/index.html) which includes sections e.g.
+- [Quickly extract overview of a given project](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/quick-overview.html)
+- [Get ideas for new fuzz targets](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/get-ideas-for-new-targets.html)
+- [Comparing introspector reports](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/comparing-introspector-reports.html)
+
+## Run Fuzz Introspector locally
+
+To generate a Fuzz Introspector report locally you can use `infra/helper.py`, which
+offers an `introspector` command. Fuzz Introspector relies on code coverage to
+analyse a given project, and this means we need to extract code coverage in the
```

Comment:
Analyze. We prefer american spelling https://developers.google.com/style/spelling